### PR TITLE
Fix decrementing of commentCount on comment deletion

### PR DESCRIPTION
**Pull request by jpaddison3**
And add a logger for denormalized updater callbacks.

Comment deletion goes through a custom resolver. The resolver was previously calling `Comments.rawUpdateOne`, this PR converts it to use updateMutator, which ensures it calls all the callbacks registered on the Comments collection.

This caused `deletedReason` to no longer be optional. From my experience, it's valuable to have people write the reason why they're deleting their own comments, so I decided to leave it required, as a callback intended, and not as the frontend claimed.

### CODE REVIEW

File: packages/lesswrong/server/resolvers/commentResolvers.ts [hunk 1]
```diff
@@ -29,10 +30,15 @@ const specificResolvers = {
           set.deletedReason = "";
           set.deletedByUserId = null;
         }
-        let modifier = { $set: set };
-        await context.Comments.rawUpdateOne({_id: commentId}, modifier);
-        const updatedComment = await context.Comments.findOne(commentId)
-        await moderateCommentsPostUpdate(updatedComment!, currentUser);
```
Comment (jpaddison3) [inline packages/lesswrong/server/resolvers/commentResolvers.ts:35]: This function is a callback that will be run on comments update.

**Comment by jpaddison3** [comment]
@s-cheng tests should be fixed

**Comment by jpaddison3** [comment]
Goddamnit

**Comment by Raemon** [comment]
Hmm, I don't actually have a strong take on whether deleting comments should require reasons, but we previously made a pretty big deal about them not requiring them. I'd want to hold off on merging this unless the requirement were forum-gated.

https://www.lesswrong.com/posts/5Ym7DN6h877eyaCnT/meta-tations-on-moderation-towards-public-archipelago

**Comment by jpaddison3** [comment]
@s-cheng lol look at this:

```
  1) Moderators
       can ban users and remove their content from public view:
     Error: The following error originated from your application code, not from Cypress.

  > Hydration failed because the initial UI does not match what was rendered on the server.
```

The tests were just failing because of a flaky error. It just happened to be on the same exact test as would have failed if there had been a real problem. (@jimrandomh FYI)

**Comment by jpaddison3** [comment]
@Raemon alright, 🤷‍♂️. Thanks for noticing. It does seem like that post might be compatible with the way the callbacks were written, as indeed they were written by a LW dev. But I don't care that much.

**Pull request merged**